contracts/token/Token.sol

```diff
@@ -385,6 +387,15 @@ contract Token is IToken, AgentRoleUpgradeable, TokenStorage {
         revert RecoveryNotPossible();
     }
 
+    /// @dev See {IToken-setAllowanceForAll}.
+    function setAllowanceForAll(bool _allow, address _caller, address[] calldata _targets) external override onlyAgent {
+        uint256 targetsCount = _targets.length;
+        for (uint256 i = 0; i < targetsCount; i++) {
+            _defaultAllowances[_caller][_targets[i]] = _allow;
+            emit DefaultAllowance(_caller, _targets[i], _allow);
+        }
+    }
+
```

Comment:
we should provide a function for token holders to get out of the default allowance if they want to handle allowance at their level, keeping more control this way